src/api/documents/collection-ref.ts

```diff
@@ -26,43 +26,62 @@ import { DocumentRef } from './document-ref';
  */
 export class CollectionRef<T extends { [key: string]: any }> {
   private documentClient: DocumentServiceClient;
-  private _collection: Collection;
+  public readonly name: string;
+  public readonly parent?: DocumentRef<any>;
 
   constructor(
     documentClient: DocumentServiceClient,
     name: string,
-    parentKey?: Key
+    parent?: DocumentRef<any>,
   ) {
     this.documentClient = documentClient;
+    this.name = name;
+    this.parent = parent;
+  }
 
-    const collection = new Collection();
-    collection.setName(name);
-
-    if (parentKey) {
-      collection.setParent(parentKey);
-    }
-
-    this._collection = collection;
+  /**
+   * Return a collection group reference from this collection
+   * @param name 
+   */
+  public collection(name: string): CollectionGroupRef<T> {
+    return CollectionGroupRef
+      .fromCollectionRef(this, this.documentClient)
+      .collection(name);
   }
 
   /**
    * Return a reference to a document in the collection.
    * @param documentId id the document unique id (required)
    * @returns new collection document reference
    */
-  public doc(documentId: string) {
+  public doc(id: string): DocumentRef<T> {
     return new DocumentRef<T>(
       this.documentClient,
-      this._collection,
-      documentId
+      this,
+      id
     );
   }
 
   /**
    * Create a new collection query object
    * @returns a new collection query object
    */
-  public query() {
-    return new Query<T>(this.documentClient, this._collection);
+  public query(): Query<T> {
+    return new Query<T>(this.documentClient, this);
+  }
+
+  /**
+   * 
+   */
```

Comment:
Doc comment needed if this is public. It would be ideal to leave this private though if possible

Reply:
done